php-frontend/src/main/java/org/sonar/php/tree/impl/statement/UseStatementTreeImpl.java

```diff
@@ -20,45 +20,73 @@
 import org.sonar.php.tree.impl.SeparatedListImpl;
 import org.sonar.php.tree.impl.lexical.InternalSyntaxToken;
 import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.NamespaceNameTree;
 import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
 import org.sonar.plugins.php.api.tree.statement.UseClauseTree;
 import org.sonar.plugins.php.api.tree.statement.UseStatementTree;
 import org.sonar.plugins.php.api.visitors.VisitorCheck;
 
-import javax.annotation.Nullable;
-import java.util.Iterator;
-
 public class UseStatementTreeImpl extends PHPTree implements UseStatementTree {
 
   private final Kind kind;
-  private InternalSyntaxToken useToken;
-  private InternalSyntaxToken useTypeToken;
-  private SeparatedListImpl<UseClauseTree> declarations;
-  private InternalSyntaxToken eosToken;
-
-  public UseStatementTreeImpl(
-      InternalSyntaxToken useToken,
-      @Nullable InternalSyntaxToken useTypeToken,
-      SeparatedListImpl<UseClauseTree> declarations,
-      InternalSyntaxToken eosToken
+  private final InternalSyntaxToken useToken;
+  private final InternalSyntaxToken useTypeToken;
+  private final NamespaceNameTree prefix;
+  private final InternalSyntaxToken nsSeparatorToken;
+  private final InternalSyntaxToken openCurlyBraceToken;
+  private final SeparatedListImpl<UseClauseTree> clauses;
+  private final InternalSyntaxToken closeCurlyBraceToken;
+  private final InternalSyntaxToken eosToken;
+
+  private UseStatementTreeImpl(
+    Tree.Kind kind,
+    InternalSyntaxToken useToken,
+    @Nullable InternalSyntaxToken useTypeToken,
+    @Nullable NamespaceNameTree prefix,
+    @Nullable InternalSyntaxToken nsSeparatorToken,
+    @Nullable InternalSyntaxToken openCurlyBraceToken,
+    SeparatedListImpl<UseClauseTree> clauses,
+    @Nullable InternalSyntaxToken closeCurlyBraceToken,
+    InternalSyntaxToken eosToken
   ) {
     this.useToken = useToken;
     this.useTypeToken = useTypeToken;
-    this.declarations = declarations;
+    this.prefix = prefix;
+    this.nsSeparatorToken = nsSeparatorToken;
+    this.openCurlyBraceToken = openCurlyBraceToken;
+    this.clauses = clauses;
+    this.closeCurlyBraceToken = closeCurlyBraceToken;
     this.eosToken = eosToken;
 
-    if (useTypeToken == null) {
-      this.kind = Kind.USE_STATEMENT;
-    } else if (useTypeToken().text().equals(PHPKeyword.CONST.getValue())) {
-      this.kind = Kind.USE_CONST_STATEMENT;
-    } else {
-      this.kind = Kind.USE_FUNCTION_STATEMENT;
-    }
+    this.kind = kind;
+  }
+
+  public static UseStatementTreeImpl createUseStatement(
+    InternalSyntaxToken useToken,
+    @Nullable InternalSyntaxToken useTypeToken,
+    SeparatedListImpl<UseClauseTree> clauses,
+    InternalSyntaxToken eosToken
+  ) {
+    return new UseStatementTreeImpl(Kind.USE_STATEMENT, useToken, useTypeToken, null, null, null, clauses, null, eosToken);
+  }
+
+  public static UseStatementTreeImpl createGroupUseStatement(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00107)